crates/relayer/src/link/relay_path.rs

```diff
@@ -1381,11 +1389,55 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
         dst_info: &ChainStatus,
         height: Height,
     ) -> Result<(Option<Any>, Option<Any>), LinkError> {
+        crate::time!(
+            "build_recv_or_timeout_from_send_packet_event",
+            {
+                "src_channel": event.packet.source_channel,
+                "dst_channel": event.packet.destination_channel,
+            }
+        );
         let timeout = self.build_timeout_from_send_packet_event(event, dst_info)?;
         if timeout.is_some() {
             Ok((None, timeout))
         } else {
-            Ok((self.build_recv_packet(&event.packet, height)?, None))
+            match event
+                .packet
+                .are_fields_valid(self.max_memo_size, self.max_receiver_size)
```

Comment:
Why are we not checking this earlier, i.e. for timeouts also? Same question for `build_ack_from_recv_event()`? Maybe even in the supervisor?

Reply:
Thinking about it more, while it would be nice to have this in the supervisor it would not cover the startup and periodic clearing. So I guess we can check in `relay_path` but let's decide if we want to also apply for the other (than recv) message types.

Reply:
The check is now performed directly in `generate_operational_data` for all events which pertain to an ICS-04 packet: https://github.com/informalsystems/hermes/pull/3765/commits/3d5687eb534dc72a94c08e95e7afdfb6492d3d22